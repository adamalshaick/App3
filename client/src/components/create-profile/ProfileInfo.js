import React from "react";
import TextFieldGroup from "../common/TextFieldGroup";
import TextAreaFieldGroup from "../common/TextAreaFieldGroup";
import UploadFileGroup from "../common/UploadFileGroup";

const ProfileInfo = ({
  errors,
  handle,
  location,
  bio,
  onChange,
  fileSelectedHandler
}) => {
  return (
    <>
      <div className="row">
        <div className="col-md-6 mb-2">
          <UploadFileGroup
            error={errors.file}
            icon="fas fa-file-upload fa-8x"
            type="file"
            name="file"
            onChange={fileSelectedHandler}
            info="Profile Image"
          />
        </div>
        <div className="col-md-6 mt-5">
          <TextFieldGroup
            id="handle"
            placeholder="* Username"
            name="handle"
            value={handle}
            onChange={onChange}
            error={errors.handle}
            info="Your username"
          />
          <TextFieldGroup
            id="location"
            placeholder="Location"
            name="location"
            value={location}
            onChange={onChange}
            error={errors.location}
            info="Your Location"
          />
        </div>
      </div>
      <TextAreaFieldGroup
        id="bio"
        placeholder="Short Bio"
        name="bio"
        value={bio}
        onChange={onChange}
        error={errors.bio}
        info="Your bio"
      />
    </>
  );
};

export default ProfileInfo;
