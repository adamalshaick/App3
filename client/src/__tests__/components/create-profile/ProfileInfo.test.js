import React from "react";
import { shallow } from "enzyme";
import ProfileInfo from "../../../components/create-profile/ProfileInfo";
import toJson from "enzyme-to-json";

const Props = {
  errors: {}
};

const wrapper = shallow(<ProfileInfo {...Props} />);

describe("render component", () => {
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
