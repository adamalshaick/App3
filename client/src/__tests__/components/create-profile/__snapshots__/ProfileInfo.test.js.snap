// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<Fragment>
  <div
    className="row"
  >
    <div
      className="col-md-6 mb-2"
    >
      <UploadFileGroup
        icon="fas fa-file-upload fa-8x"
        info="Profile Image"
        name="file"
        type="file"
      />
    </div>
    <div
      className="col-md-6 mt-5"
    >
      <InputGroup
        id="handle"
        info="Your username"
        name="handle"
        placeholder="* Username"
        type="text"
      />
      <InputGroup
        id="location"
        info="Your Location"
        name="location"
        placeholder="Location"
        type="text"
      />
    </div>
  </div>
  <TextAreaFieldGroup
    id="bio"
    info="Your bio"
    name="bio"
    placeholder="Short Bio"
  />
</Fragment>
`;
